tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -192,6 +194,43 @@ def ldap_auth_name():
     yield ldap_name
 
 
+def run_command(cmd, hostname=settings.server.hostname, timeout=None):
+    """helper function for ssh command and avoiding the return code check in called function"""
+    if timeout is not None:
+        result = ssh.command(cmd=cmd, hostname=hostname, timeout=timeout)
+    else:
+        result = ssh.command(cmd=cmd, hostname=hostname)
+    if result.return_code != 0:
+        raise CLIReturnCodeError(
+            result.return_code,
+            result.stderr,
+            'Failed to run the command : {}'.format(cmd),
+        )
+    else:
+        return result.stdout
+
+
+@fixture(scope='module')
+def enroll_idm_and_configure_external_auth():
+    """Enroll the Satellite6 Server to an IDM Server."""
+    run_command(cmd='yum -y --disableplugin=foreman-protector install ipa-client ipa-admintools')
+
+    run_command(cmd='echo {0} | kinit admin'.format(settings.ipa.password_ipa),
+                hostname=settings.ipa.hostname_ipa)
+    result = run_command(cmd='ipa host-add --random {}'.format(settings.server.hostname),
+                         hostname=settings.ipa.hostname_ipa)
+
+    password = result[4][-22:]
```

Comment:
I don't know output of `ipa host-add`, but could this be made a little less magic-numbery? Possibly loop over entire thing to find line matching some pattern, and use `split` to extract part we need?

Reply:
@mirekdlugosz As suggested applied some better logic